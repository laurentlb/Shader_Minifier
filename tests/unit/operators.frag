#version 330

void main()
{
  int a = 1 + 2 * 3 + 4;
  int b = 14 / 2;
  int c = 8 % 3;
  
  float f = 1.23456 * 2.1234;
  f = 1.4 / 2.;
  f = 1.4 / 3.;
  float g = mod(8., 3.);
  
  float z =
    (1.  < 2. ? 1 : 2) +
    (1.  > 2. ? 1 : 2) * 10 +
    (1. <= 2. ? 1 : 2) * 100 +
    (1. >= 2. ? 1 : 2) * 1000 +
	(1. == 2. ? 1 : 2) * 10000 +
	(1. != 2. ? 1 : 2) * 100000;
}

int no_parens(int a, int b, int c) {
	return a + (b + c) + a + (b * c);
}

int no_parens2(int a, int b, int c) {
	int d = a*(b*c);
	return a - (b - 1) - (d - c);
}

int other(int a, int b, int c, int d) {
    return (a*b)*(c*d)*(a*b);
}
