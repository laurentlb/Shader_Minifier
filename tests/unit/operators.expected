// Generated with  (https://github.com/laurentlb/Shader_Minifier/)
#ifndef SHADER_MINIFIER_IMPL
#ifndef SHADER_MINIFIER_HEADER
# define SHADER_MINIFIER_HEADER
#endif

#else // if SHADER_MINIFIER_IMPL

// tests/unit/operators.frag
 "#version 330\n"
 "float notmain()"
 "{"
   "int a=11,b=7,c=2;"
   "float f=1.4/3.,g=mod(8.,3.),z=122121;"
   "return a+b+c+f+g+z;"
 "}"
 "int no_parens(int a,int b,int c)"
 "{"
   "return b+c+a+a+b*c;"
 "}"
 "int no_parens2(int a,int b,int c)"
 "{"
   "int d=b*c*a;"
   "return a-b+1-d+c;"
 "}"
 "int other(int a,int b,int c,int d)"
 "{"
   "return a*b*(c*d)*(a*b);"
 "}"
 "float f(float x)"
 "{"
   "float a=(x+=1.,length(vec3(x)));"
   "return a*a*sin((a*=a,a));"
 "}",

#endif
